Fix dayview glitches (rebase of #692)

SebiderSushi opened the pull request:
Fixes #457. #693 
Merges #692 
TODO: The commit messages might discard too much information and do not link original commits.

This is a PR from what has been my current master branch.  
It enables code review while still being the most verbose and true to history, i.e. no commits have been melded yet, it's just the cherry-picks.  
My suggestion is to go from here. Whatever will be decided for the final squash that should be merged, i would definitely change the commit messages a bit to reflect on the squashed changes which, as i know now, are quite intertwined.

## Summary of my squashing considerations

In my opinion, https://github.com/SebiderSushi/Etar-Calendar/commits/squashed is actually the least clean of my branches and definitely not atomic. I also squashed `Fix problem with selecting hour 23` into `Fix issue #457 and remove other bounces` while keeping `Don't mess with selected hour when scrolling`. As i found out until now, doing so is incorrect:

> I did some testing and apparently,
> `dbfed92215570932c7acb4d2d2b4aa7b4e255e3a Fix problem with selecting hour 23`
> only works if applied together with
> `8f59dc15ed7fc97edcac7d1167696e9aa8dc27b0 Don't mess with selected hour when scrolling`
> 
> Now i am very confused as to how to squash of retain the commits of this PR.

Concerning this issue, @rparkins999 [stated](https://github.com/Etar-Group/Etar-Calendar/issues/693#issuecomment-627263825_) in #693:
> The problem with setting hour 23 only occurs when the display is squeezed so that all hours are visible. It is present both before and after my recent changes, but it is worse after.

I was not able to reproduce this on my device. During my tests, i could easily reproduce the issue after applying `Fix issue #457 and remove other bounces`, but never reproduced it without this commit.  
If others are able to reproduce this as stated by @rparkins999 then i don't know, but according to my tests this issue is heavily linked to `Fix issue #457 and remove other bounces` so i'd suggest squashing the 3 commits together along with a descriptive commit description.

The only commit that i would possibly preserve as an individual commit is `Display hour 00 label if there is room for it.` as it can be seen to resemble one distinct feature, is atomic and could (in theory) stand on its own (which i didn't test practically).

## Original commits used on this branch
Right now, this branch contains the following, cherry-picked and unchanged commits from https://github.com/rparkins999/Etar-Calendar:
```
dbfed92215570932c7acb4d2d2b4aa7b4e255e3a Fix problem with selecting hour 23
8f59dc15ed7fc97edcac7d1167696e9aa8dc27b0 Don't mess with selected hour when scrolling
cd1ecb86bba786d5d7905319298dad1af0990375 Display hour 00 label if there is room for it.
b043785f79d46e6f24035ad0b6a51c5216bae41d Remove DEBUG_SCROLLING, left in branch DEBUG_SCROLLING
36f1e6609eea42a4a8c5ea942983e577dea4a484 Fix issue #457 and remove other bounces
```
Cherry-picking command line to reproduce:
`git cherry-pick 36f1e66 b043785 cd1ecb8 8f59dc1 dbfed92`

jspricke commented on src/com/android/calendar/DayView.java:
```diff
@@ -4005,13 +4026,14 @@ private void doScroll(MotionEvent e1, MotionEvent e2, float deltaX, float deltaY
                 }
             }
 
-            if (mViewStartY < 0) {
```
these changes here seems to be only cosmetic, can you revert them?

jspricke commented on src/com/android/calendar/DayView.java:
```diff
@@ -4166,16 +4186,20 @@ public boolean onScale(ScaleGestureDetector detector) {
+            mMaxViewStartY = 0;
+        }
 
-        if (mViewStartY < 0) {
```
dito here

jspricke commented:
There seem to be three independent fixes in here:  #457. #693 and drawing 00:00. @SebiderSushi are you willing to separate them or should I come up with commits?

SebiderSushi commented:
I think these three are good semantical units. My only concerms are that issue #693 is intertwined with the fix to #457 and that there are different logical intentions between `Fix problem with selecting hour 23` and `Don't mess with selected hour when scrolling`.

If that's okay for you then i'm fine with squashing together three commits as
```
Fix #457 <and details>
Fix #693 <and details>
00-hour label
```

jspricke commented:
Sounds good.
For completeness: I think a fix for #457 would be accepted. For #693 there is actually a function to do that on purpose: https://github.com/Etar-Group/Etar-Calendar/blob/master/src/com/android/calendar/DayView.java#L1870 so we would need to discuss what is more intuitive or other options and for 00:00 I'm actually not sure, as we don't print 24:00 either.

SebiderSushi commented:
Yeah the problem with seperating the fixes for #457 and #693 is that after applying just the fix for #457 on my device, the hour marked 23:00 cannot be selected at all. Thus, the fix introduces a new issue on my device which seems to be intended to be fixed by `Fix problem with selecting hour 23` but during my tests, this fix was ineffective as long as the commit `Don't mess with selected hour when scrolling.` was not applied as well.

This is what i'm trying to explain with all my "squashing considerstions". It's that only the 00-hour label commit could (maybe, not tested!) stand on it's own but at the same time each commit correlates to some semantical unit.

Maybe someone with deeper knowledge of the codebase could resolve this or @rparkins999 could have another look at this but creating atomic and independent commits out of this is beyond my abilities right now and maybe not even worth the effort.


Apart from the squashing questions i do support the changes that drop the resetting of the selected hour. I like the bahavior of the DayView with these changes applied. The commit in Line 1870 that you linked does not make sense to me, as the cureent Etar build version 1.0.21 on F-Droid removes the hour selection as soon as i touch the DayView.
Also, as i mentioned in #683, i think that the selected hour should be completely unlinked from the FAB to resolve all other issues i can see here.

Anyways, i'll be off for the weekend. If you can find a solution and are getting impatient feel free to squash around my own commits while i'm gone. If i didn't make any mistake, my two fixup commits are meant to be squashed into the `Fix issue #457 and remove other bounces` commit.

SebiderSushi commented:
I think it's ok if the selected hour can be scrolled out of the viewframe by the user, since you can clearly see it moving out of the screen, still selected.

jspricke commented:
>Yeah the problem with seperating the fixes for #457 and #693 is that after applying just the fix for #457 on my device, the hour marked 23:00 cannot be selected at all. Thus, the fix introduces a new issue on my device which seems to be intended to be fixed by `Fix problem with selecting hour 23` but during my tests, this fix was ineffective as long as the commit `Don't mess with selected hour when scrolling.` was not applied as well.

Took me some time to get this, I assume you are talking about the row 
after the 23:00 line. I agree that this would be a bug.

>This is what i'm trying to explain with all my "squashing considerstions". It's that only the 00-hour label commit could (maybe, not tested!) stand on it's own but at the same time each commit correlates to some semantical unit.
>
>Maybe someone with deeper knowledge of the codebase could resolve this or @rparkins999 could have another look at this but creating atomic and independent commits out of this is beyond my abilities right now and maybe not even worth the effort.

Agreed, probably not worth the effort just for the label. Let's wait if 
anyone else rises concerns and then leave it in.

>Apart from the squashing questions i do support the changes that drop the resetting of the selected hour. I like the bahavior of the DayView with these changes applied. The commit in Line 1870 that you linked does not make sense to me, as the cureent Etar build version 1.0.21 on F-Droid removes the hour selection as soon as i touch the DayView.
>Also, as i mentioned in #683, i think that the selected hour should be completely unlinked from the FAB to resolve all other issues i can see here.

I'm ok both ways, could you remove the `resetSelectedHour()` if we 
remove all of it's calls?

SebiderSushi commented:
Huh, it totally slipped me but if `resetSelectedHour()` becomes a dead method it should be taken out of course.

As for the final commits i will definitely squash away the fixups and keep the 00-hour label commit as-is, but i am unsure if i should squash or keep the commits related to

jspricke commented:
>Huh, it totally slipped me but if `resetSelectedHour()` becomes a dead method it should be taken out of course.

Could you add a commit to delete it?

>As for the final commits i will definitely squash away the fixups and keep the 00-hour label commit as-is, but i am unsure if i should squash or keep the commits related to

I would keep the 00-hour commit and add a comment that it corrects the 
selection but introduced before.

SebiderSushi commented:
Just did some testing and the `Display hour 00 label if there is room for it.` commit works as expected if cherrypicked alone onto the master branch.

SebiderSushi commented:
Okay i just force pushed the general concept of the sqashed commits.
The commit messages are incomplete, i am especially unsure about how to document the relation to the original commits. (Any style you prefer?)

### git rebase-todo:
```
pick 2a90cd4c Fix issue #457 and remove other bounces
squash 77eb9ab6 Remove DEBUG_SCROLLING, left in branch DEBUG_SCROLLING
squash e011aa2b fixup: undo out of scope changes
squash 9c878523 fixup: undo non-functional if block reorganizations
pick 7ef8b31a Don't mess with selected hour when scrolling
squash 212a2448 Fix problem with selecting hour 23
squash 5d200975 fixup: remove now unused removeSelectedHour()
pick a267051b Display hour 00 label if there is room for it.
```

jspricke commented:
> The commit messages are incomplete, i am especially unsure about how to document the relation to the original commits. (Any style you prefer?)

What do you mean by incomplete? Anything specific you want to add?

I would be fine with this, thanks for working on it :).

SebiderSushi commented:
Now i consider these commits formally complete and sign them off to merge if all went well and they work as expected.

I have added the warning about introducing a bug with the first commit and especially mentions of the original commits that i used, as i feel the need to keep that history linked for as long as these commits exist. This is probably mostly a personal thing to me as i find it quite weird to suqash commits of others, so i guess i feel the need to put any attribution there as i've took the original commits and changed them in some way. This whole thing should be as transparent as possible in my opinion.

But of course, if you know any guidelines about commit squashing or have a project policy about it i would readily follow any of it.

The messages look kind of messy now but i can't think of any prettier way of formatting them.
They also contain a lot of information but i was thinking to better write a little too much documentation no one will ever read than to have this information missing if someone might ever be looking for it.

jspricke commented:
>I have added the warning about introducing a bug with the first commit and especially mentions of the original commits that i used, as i feel the need to keep that history linked for as long as these commits exist. This is probably mostly a personal thing to me as i find it quite weird to suqash commits of others, so i guess i feel the need to put any attribution there as i've took the original commits and changed them in some way. This whole thing should be as transparent as possible in my opinion.
>
>But of course, if you know any guidelines about commit squashing or have a project policy about it i would readily follow any of it.

The only thing I'm aware of would be Co-developed-by: as specified here:

https://www.kernel.org/doc/html/latest/process/submitting-patches.html#when-to-use-acked-by-cc-and-co-developed-by

Feel free to use it but we don't have a project policy for it.

Two small style nitpicks:

https://github.com/Etar-Group/Etar-Calendar/pull/710/commits/e8190812e267cc5fc961a4a58f2a1818200c70fe

`Adresses <#693>`

Should be:

Closes: #693

According to: https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords

https://github.com/Etar-Group/Etar-Calendar/pull/710/commits/46b47658c9be8646ef91ff9df528273f6f9f83b6

drop the "This is a verbatim cherrypick of <rparkins999@cd1ecb8>"
This is already stated by the Author header.

SebiderSushi commented:
I am content with this know. Still any remarks?

jspricke commented:
Thanks a lot!
